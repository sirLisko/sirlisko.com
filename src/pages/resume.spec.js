import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import Resume from "./resume.js";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `I am the title`,
        },
      },
    })
  );
});

describe("Resume Page", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = renderer.create(<Resume />).toJSON();
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
