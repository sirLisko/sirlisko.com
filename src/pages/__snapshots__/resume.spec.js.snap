// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume Page should render properly 1`] = `
<div
  className="resume"
>
  <section
    className="me"
  >
    <div
      className="me__face"
    >
      <a
        href="/"
      >
        <img
          alt="sirlisko face"
          src="https://sirlisko.com/images/sirlisko.svg"
        />
      </a>
    </div>
    <p
      className="me__intro"
    >
      <span>
        Luca Lischetti
      </span>
      <span>
        a.k.a. sirLisko
      </span>
      <span>
        Senior Software Engineer
      </span>
    </p>
  </section>
  <div
    className="wrapper"
  >
    <article
      className="skills"
    >
      <h1>
        Skills
      </h1>
      <div>
        <h2>
          main
        </h2>
        <ul>
          <li>
            Javascript
          </li>
          <li>
            React
          </li>
          <li>
            Node
          </li>
          <li>
            Markup/CSS
          </li>
          <li>
            Front End Architecture
          </li>
          <li>
            FE Build & CI
          </li>
          <li>
            BDD/TDD
          </li>
          <li>
            Functional Testing
          </li>
        </ul>
      </div>
      <div>
        <h2>
          misc.
        </h2>
        <ul>
          <li>
            Agile / Scrum
          </li>
          <li>
            Redux
          </li>
          <li>
            GraphQL
          </li>
          <li>
            Typescript
          </li>
          <li>
            Styled-Components
          </li>
          <li>
            Webpack
          </li>
          <li>
            NextJS
          </li>
          <li>
            Gatsby
          </li>
          <li>
            JAM stack
          </li>
          <li>
            Gulp / Grunt
          </li>
          <li>
            Chrome Extensions
          </li>
          <li>
            Bash Scripting
          </li>
          <li>
            User Scripting
          </li>
          <li>
            Web Scraping
          </li>
        </ul>
      </div>
    </article>
    <article
      className="section experiences"
    >
      <h1>
        Experiences
      </h1>
      <div>
        <h2>
          Kalo
           
          <span>
            04.19 - 06.19
          </span>
        </h2>
        <p>
          Analysed, standardized and improved the whole Front End, with a main focus on the common UI library.
        </p>
        <p>
          Architected and created an MVP for a Front End Monorepo.
        </p>
        <p>
          Worked on improving performances and the Dev experience.
        </p>
      </div>
      <div>
        <h2>
          Reason
           
          <span>
            09.18 - 12.18
          </span>
        </h2>
        <p>
          Developed from scratch the whole Front End and some basic services for 3rd party client Coverly.
        </p>
        <p>
          Worked with React/Redux, Jest for testing, Storybook for documentation, and Node for API proxy.
        </p>
        <p>
          Deployed to AWS and Azure using Bitbucket pipelines.
        </p>
        <p>
          Dealt with the client and the API provider. Iterated on product and design to facilitate the development.
        </p>
      </div>
      <div>
        <h2>
          Verve
           
          <span>
            05.17 - 07.18
          </span>
        </h2>
        <p>
          Worked on improving the user experience through the check-out with technologies involving React/Redux, Relay, and GraphQL.
        </p>
        <p>
          Improved the Front End unit testing environment, migrated to Jest and introduced snapshot testing.
        </p>
        <p>
          Integrated 3rd party payment methods (Stripe).
        </p>
        <p>
          Worked with the design team to improve the UI/UX.
        </p>
      </div>
      <div>
        <h2>
          Pobble Education
           
          <span>
            06.16 - 05.17
          </span>
        </h2>
        <p>
          Rewrote the whole Front End architecture following the Angular best practices, decoupling it from Rails.
        </p>
        <p>
          Introduced a Front End build system. Defined a code style guide. Introduced unit/e2e test frameworks.
        </p>
      </div>
      <div>
        <h2>
          Shazam Entertainment
           
          <span>
            10.11 - 05.16
          </span>
        </h2>
        <p>
          Developed the main website (shazam.com).
        </p>
        <p>
          Co-operated in the realization of a modular architecture of the Front End (AMD, BEM) and its build system (Node and Gulp).
        </p>
        <p>
          Developed a custom solution for functional tests.
        </p>
        <p>
          Worked on automation tools and localisations.
        </p>
      </div>
      <div>
        <h2>
          YOOX Net-a-Porter Group
           
          <span>
            09.10 - 10.11
          </span>
        </h2>
        <p>
          Mainly involved in JavaScript stuff, especially in testing TDD, unit and functional, continuous integration, and build automation.
        </p>
        <p>
          Projected and developed a targeted JS framework.
        </p>
      </div>
      <div>
        <h2>
          Once upon a time...
           
        </h2>
        <p>
          I had the pleasure to work also with Java, Scala, PHP, MySQL, C++, Ruby on Rails. I don't consider myself an expert on those languages but I like, especially in my free time, playing with other technologies :)
        </p>
      </div>
    </article>
    <article
      className="section projects"
    >
      <h1>
        Side Projects
      </h1>
      <div>
        <h2>
          <a
            href="https://been.netlify.com/"
          >
            Been
          </a>
        </h2>
        <p>
          Web companion to keep track of where you have BEEN, listing your visited countries.
        </p>
        <p>
          React app, with auth and databased powered by Firebase. Styled using Styled-Components. And a handcrafted World Map SVG.
        </p>
        <p>
          The project is still Work in Progress :)
        </p>
      </div>
      <div>
        <h2>
          <a
            href="https://shouldilistenit.sirlisko.com/"
          >
            Should I listen it?
          </a>
        </h2>
        <p>
          Little webapp that shows you the most played songs by an artist in the last two years of gigs.
        </p>
        <p>
          Node.js app that proxies external APIs (Setlist.fm, Spotify and Songkick) and mashes them up to set you up for your next Gig.
        </p>
      </div>
      <div>
        <h2>
          <a
            href="https://chrome.google.com/webstore/detail/zamify/foilfgbdcipbajipeodmjjnkflkfocin"
          >
            Shazamify
          </a>
        </h2>
        <p>
          Shazamify allows you to play your Shazams in Spotify, directly from the browser, by adding a small Spotify icon to every Shazam track page.
        </p>
        <p>
          Chrome Extension powered by Spotify OAuth and Web API.
        </p>
      </div>
      <div>
        To see a full list of my side projects go to
         
        <a
          href="/projects"
        >
          https://sirlisko.com/projects
        </a>
        .
      </div>
    </article>
    <article
      className="section education"
    >
      <h1>
        Education
      </h1>
      <p>
        Degree in Computer Science @ Università degli studi dell’Insubria & Universidad de Salamanca.
      </p>
      <p>
        Thesis on Assistive Technology -
         
        <a
          data-beacon={true}
          href="http://assistivetechnology.it"
        >
          http://assistivetechnology.it
        </a>
         
        (sorry link in Italian).
      </p>
    </article>
    <article
      className="section extras"
    >
      <h1>
        Extras
      </h1>
      <p>
        I co-held a workshop about TDD in Javascript @ Università di Cesena.
      </p>
      <p>
        I’m a Volunteer and occasional collaborator of the Front-End Conference -
         
        <a
          data-beacon={true}
          href="https://fromthefront.it"
        >
          From The Front
        </a>
        .
      </p>
      <p>
        I have a Cisco CCNA cetificate (Cisco Certified Network Associate).
      </p>
      <p>
        I have a training certificate as Fire Marshal.
      </p>
    </article>
    <article
      className="section interests"
    >
      <h1>
        Interests
      </h1>
      <p>
        I like very much 
        <a
          href="https://www.last.fm/user/sirlisko"
        >
          music
        </a>
        ,
         
        <a
          href="https://www.imdb.com/user/ur15844029/"
        >
          cinema
        </a>
        ,
         
        <a
          href="https://www.goodreads.com/sirLisko"
        >
          books
        </a>
        , and, last but not least, 
        <a
          href="https://pizzaclub.sirlisko.com/"
        >
          pizza
        </a>
        .
      </p>
      <p>
        I like Art, especially modern and 8-bit graphic (like old-school video games), travel, languages. I’m also an active member of
         
        <a
          href="https://www.couchsurfing.com/people/sirlisko"
        >
          CouchSurfing community
        </a>
        .
      </p>
    </article>
    <article
      className="section contacts"
    >
      <h1>
        Contacts
      </h1>
      <p>
        My website 
        <a
          href="https://sirlisko.com"
        >
          https://sirlisko.com
        </a>
      </p>
      <p>
        My Github
         
        <a
          href="https://github.com/sirlisko"
        >
          https://github.com/sirlisko
        </a>
      </p>
      <p>
        My email address 
        <a
          href="mailto:luca@sirlisko.com"
        >
          luca@sirlisko.com
        </a>
        .
      </p>
      <p>
        A print-friendly version of this:
         
        <a
          href="https://sirlisko.com/cv/alt.pdf"
        >
          https://sirlisko.com/cv/alt.pdf
        </a>
      </p>
    </article>
  </div>
</div>
`;
